src/scripts/main.js

```diff
@@ -1,3 +1,35 @@
 'use strict';
 
-// write code here
+const list = document.querySelectorAll('li');
+const names = [...list].map(li => li.innerText);
+const dataSet = [...list].map(li => li.dataset);
+
+function toNumber(element) {
+  return Number(element.slice(1).split(',').join(''));
+}
+
+const dataSetWithNames = dataSet.map((args, index) => {
+  return {
+    ...args,
+    nameOfEmployee: names[index],
+  };
+});
+
+function sortList() {
+  dataSetWithNames.sort((a, b) => toNumber(b.salary) - toNumber(a.salary));
+}
+
+sortList();
```

Comment:
you should pass the list to the function as a parameter

```suggestion
function sortList(list) {
  list.sort((a, b) => toNumber(b.salary) - toNumber(a.salary));
}

sortList(dataSetWithNames);
```